out->info should respect out->is_quiet

clumens opened the pull request:
I took a survey of all the places out->info was being used and I think the only ones left are those that can safely be ignored if is_quiet is set.  But it might be worth another look.

kgaillot reviewed (commented):
I love this, I think it helps both reading and writing code flow more easily. BTW the CI failure is just travis's ppc64le host having its usual timeouts, we can ignore that.

->info() calls that still need addressing:

* tools/crm_resource_runtime.c:cli_resource_restart(): it makes sense that the "Waiting for ..." messages go away with quiet, but the "Could not complete" messages are really errors (it would be acceptable to leave display_list() as you have it, so in those cases just the error message is printed and not the list of resources). Treating those as errors will allow XML to get some useful info out of it as well.
* tools/crm_resource_runtime.c:cli_resource_execute_from_params(): I think we're OK with this one because we exit op->rc, but I'm on the fence whether the "Operation ... failed" message should be an error (especially since that would get it into XML as well).
* tools/crm_mon.c: This one is interesting because we have a --quiet argument but we don't hook it up to the output object; all it does currently is exclude the summary times. It might be worthwhile to hook it up, but then we'd need to be more selective about what we do with info() here. Most of the info() calls would be fine quieted, and that would be a nice feature, but some (like print_option_help()) would need to be changed.

clumens commented:
> I love this, I think it helps both reading and writing code flow more easily.

I like how it looks, too.  Good suggestion.

>     * tools/crm_resource_runtime.c:cli_resource_restart(): it makes sense that the "Waiting for ..." messages go away with quiet, but the "Could not complete" messages are really errors (it would be acceptable to leave display_list() as you have it, so in those cases just the error message is printed and not the list of resources). Treating those as errors will allow XML to get some useful info out of it as well.

Fixed.

>     * tools/crm_resource_runtime.c:cli_resource_execute_from_params(): I think we're OK with this one because we exit op->rc, but I'm on the fence whether the "Operation ... failed" message should be an error (especially since that would get it into XML as well).

I converted the "Operation ... failed" warning into an error.  If it has "failed" in the name, it seems like it really should be an error.  The others I've left as-is.

>     * tools/crm_mon.c: This one is interesting because we have a --quiet argument but we don't hook it up to the output object; all it does currently is exclude the summary times. It might be worthwhile to hook it up, but then we'd need to be more selective about what we do with info() here. Most of the info() calls would be fine quieted, and that would be a nice feature, but some (like print_option_help()) would need to be changed.

print_option_help should only ever appear in curses output, so there's no real reason for this to use formatted output (aside from the convenience of not needing to know anything about curses).  I could add curses_formatted_(v)printf to tools/crm_mon_curses.c and use them instead here.

As for the rest of the stuff in crm_mon.c, I could convert that in this PR or do a separate one at some point in the future.

kgaillot commented:
> print_option_help should only ever appear in curses output, so there's no real reason for this to use formatted output (aside from the convenience of not needing to know anything about curses). I could add curses_formatted_(v)printf to tools/crm_mon_curses.c and use them instead here.
> 
> As for the rest of the stuff in crm_mon.c, I could convert that in this PR or do a separate one at some point in the future.

Let's leave crm_mon for a future PR since quiet doesn't affect it currently. Good point, print_option_help() and friends can just move to crm_mon_curses.c.
